substratecommon.go

```diff
@@ -0,0 +1,815 @@
+	return (func(r *RequestOptions) {
+		r.Creator = creator
+	})
+}
+
+// WithDependentTxID allows specifying a dependency on a transaction ID.  If
+// set, the client will poll for the presence of that transaction before
+// simulating the request on the peer with the transaction.
+func WithDependentTxID(txID string) Config {
+	return (func(r *RequestOptions) {
+		r.DependentTxID = txID
+	})
+}
+
+// WithConditionalDependentTxID allows specifying a conditional dependency on a
+// transaction ID when polling is disabled or transaction dependencies are
+// already enabled.  This is intended for use with chained sequential calls that
+// have a critical dependency.
+func WithConditionalDependentTxID(txID string) Config {
+	return (func(r *RequestOptions) {
+		if r.DisableWritePolling || r.DependentTxID != "" {
+			r.DependentTxID = txID
+		}
+	})
+}
+
+// WithDisableWritePolling allows disabling polling for full consensus after a
+// write is committed.
+func WithDisableWritePolling(disable bool) Config {
+	return (func(r *RequestOptions) {
+		r.DisableWritePolling = disable
+	})
+}
+
+// WithCCFetchURLDowngrade allows controlling https -> http downgrade,
+// typically useful before proxying for ccfetchurl library.
+func WithCCFetchURLDowngrade(downgrade bool) Config {
+	return (func(r *RequestOptions) {
+		r.CCFetchURLDowngrade = downgrade
+	})
+}
+
+// WithCCFetchURLProxy sets the proxy for ccfetchurl library.
+func WithCCFetchURLProxy(proxy string) Config {
+	return (func(r *RequestOptions) {
+		r.CCFetchURLProxy = proxy
+	})
+}
+
+// FlattenOptions will flatten a list of config options.
+func FlattenOptions(configs ...Config) *ConcreteRequestOptions {
+	opt := &RequestOptions{}
+
+	for _, config := range configs {
+		config(opt)
+	}
+
+	return &ConcreteRequestOptions{
+		Headers:             opt.Headers,
+		Endpoint:            opt.Endpoint,
+		ID:                  opt.ID,
+		AuthToken:           opt.AuthToken,
+		Params:              opt.Params,
+		Transient:           opt.Transient,
+		Timestamp:           opt.TimestampGenerator(opt.Ctx),
+		MSPFilter:           opt.MSPFilter,
+		MinEndorsers:        opt.MinEndorsers,
+		Creator:             opt.Creator,
+		DependentTxID:       opt.DependentTxID,
+		DisableWritePolling: opt.DisableWritePolling,
+		CCFetchURLDowngrade: opt.CCFetchURLDowngrade,
+		CCFetchURLProxy:     opt.CCFetchURLProxy,
+	}
+}
+
+// Error represents a possible error. IsTimeoutError indicates whether
+// the error was a timeout error.
+type Error struct {
+	IsTimeoutError bool
+	Diagnostic     string
+}
+
+func (e Error) Error() string {
+	return e.Diagnostic
+}
+
+// Response represents a shiroclient response.
+type Response struct {
+	ResultJSON    []byte
+	HasError      bool
+	ErrorCode     int
+	ErrorMessage  string
+	ErrorJSON     []byte
+	TransactionID string
+}
+
+// Transaction represents summary information about a transaction.
+type Transaction struct {
+	ID     string
+	Reason string
```

Comment:
These have been  updated to include the chaincode ID and the events. Please see latest APIs and make sure this is at parity.